code/backend/batch/utilities/helpers/EnvHelper.py

```diff
@@ -9,12 +10,14 @@
 
 class EnvHelper:
     _instance = None
+    _lock = threading.Lock()
 
     def __new__(cls):
-        if cls._instance is None:
-            cls._instance = super(EnvHelper, cls).__new__(cls)
-            cls._instance.__load_config()
-        return cls._instance
+        with cls._lock:
+            if cls._instance is None:
+                cls._instance = super(EnvHelper, cls).__new__(cls)
+                cls._instance.__load_config()
+            return cls._instance
```

Comment:
From what I've seen online, it seems to be more like this

```suggestion
        if cls._instance is None:
            with cls._lock:
                if cls._instance is None:
                    cls._instance = super(EnvHelper, cls).__new__(cls)
                    cls._instance.__load_config()
        return cls._instance
```

https://medium.com/analytics-vidhya/how-to-create-a-thread-safe-singleton-class-in-python-822e1170a7f6

Reply:
The problem with that, is that the `cls._instance` for the second thread will **not** be none, so it will return it without entering the lock and waiting for the first thread to load the config.

Reply:
Ah, so we haven't solved anything 😅